tsl/src/hypercore/arrow_cache_explain.c

```diff
@@ -106,33 +75,45 @@ explain_decompression(Query *query, int cursorOptions, IntoClause *into, Explain
 	standard_ExplainOneQuery(query, cursorOptions, into, es, queryString, params, queryEnv);
 	if (decompress_cache_print)
 	{
-		Assert(es->format < sizeof(format_texts) / sizeof(*format_texts));
-
-		ExplainOpenGroup("Array cache", "Arrow Array Cache", true, es);
-		ExplainPropertyInteger(format_texts[es->format].hits_text,
-							   NULL,
-							   decompress_cache_stats.hits,
-							   es);
-		ExplainPropertyInteger(format_texts[es->format].miss_text,
-							   NULL,
-							   decompress_cache_stats.misses,
-							   es);
-		ExplainPropertyInteger(format_texts[es->format].evict_text,
-							   NULL,
-							   decompress_cache_stats.evictions,
-							   es);
-		ExplainPropertyInteger(format_texts[es->format].decompress_text,
-							   NULL,
-							   decompress_cache_stats.decompressions,
-							   es);
-
-		if (es->verbose)
-			ExplainPropertyInteger(format_texts[es->format].decompress_calls_text,
-								   NULL,
-								   decompress_cache_stats.decompress_calls,
-								   es);
-
-		ExplainCloseGroup("Array cache", "Arrow Array Cache", true, es);
+		const bool has_decompress_data = decompress_cache_stats.decompressions > 0 ||
+										 decompress_cache_stats.decompress_calls > 0;
+		const bool has_cache_data = decompress_cache_stats.hits > 0 ||
+									decompress_cache_stats.misses > 0 ||
+									decompress_cache_stats.evictions > 0;
+		if (has_decompress_data || has_cache_data)
+		{
+			if (es->format == EXPLAIN_FORMAT_TEXT)
+			{
+				appendStringInfoString(es->str, "Array:");
+				if (has_cache_data)
+					appendStringInfoString(es->str, " cache");
+				append_if_positive(es->str, "hits", decompress_cache_stats.hits);
+				append_if_positive(es->str, "misses", decompress_cache_stats.misses);
+				append_if_positive(es->str, "evictions", decompress_cache_stats.evictions);
+				if (has_decompress_data)
+					appendStringInfoString(es->str, ", decompress");
+				append_if_positive(es->str, "count", decompress_cache_stats.decompressions);
+				append_if_positive(es->str, "calls", decompress_cache_stats.decompress_calls);
```

Comment:
Shouldn't this `calls` be added only when `es->verbose==true` like in on-text format?

Reply:
Rather the opposite, the non-text format should not have this field as "verbose".